### PR TITLE
MAGN-7958 Camera position isn't loaded from file after camera has been orbited upside down

### PR DESCRIPTION
**Pull request by ikeough**
http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7958

### Purpose

The purpose of this PR is to fix [MAGN-7958](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7958). This is done in the following way:

- Unifying the camera save logic for all `Watch3DViewModel` types.
- Saving/Loading the "up" vector for a camera.
- Refactoring default camera position information into a `CameraData` class which acts as a container for information about camera position and orientation for the purpose of loading cameras, saving cameras, and resetting them to default orientations.

An example of different camera orientations loading in background an Watch 3D nodes:
image

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate
  - Existing camera load/save tests are extended to include the up vector.
- [x] User facing strings, if any, are extracted into `*.resx` files
  - No user facing string required extraction.

### Reviewers

@ramramps

### CODE REVIEW

**Comment by holyjewsus** [comment]
I have nothing to say about this PR except - that is an awesomely high res screen shot. :stuck_out_tongue: :

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 1]
```diff
@@ -34,6 +37,30 @@
...
+        public double NearPlaneDistance { get; set; }
+        public double FarPlaneDistance { get; set; }
+
+        public CameraData()
+        {
+            Name = "Default Camera";
+            EyePosition = defaultCameraPosition;
+            UpDirection = defaultCameraUpDirection;
+            LookDirection = defaultCameraLookDirection;
+            NearPlaneDistance = 0.1;
```
Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: Please store this in constant.

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 2]
```diff
@@ -34,6 +37,30 @@
...
+        public Point3D EyePosition { get; set; }
+        public Vector3D UpDirection { get; set; }
+        public Vector3D LookDirection { get; set; }
+        public string Name { get; set; }
+        public double NearPlaneDistance { get; set; }
+        public double FarPlaneDistance { get; set; }
+
+        public CameraData()
+        {
+            Name = "Default Camera";
```
Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: If required, please store the string in Resource file

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 3]
```diff
@@ -311,6 +334,74 @@ public static HelixWatch3DViewModel Start(Watch3DViewModelStartupParams paramete
...
+                    Name = name,
+                    EyePosition = new Point3D(ex, ey, ez),
+                    LookDirection = new Vector3D(lx, ly, lz),
+                    UpDirection = new Vector3D(ux, uy, uz)
+                };
+
+                return camData;
+            }
+            catch (Exception ex)
+            {
```
Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: SerializeCamerahas exception logged. Should DeserializeCamera do the same?

File: src/Libraries/DynamoWatch3D/Watch3D.cs [hunk 4]
```diff
@@ -283,35 +254,24 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
...
+                    // For the node model, we need to supply the deserialization method with the camera node.
+                    var cameraElement = nodeElement.ChildNodes.Cast<XmlNode>().FirstOrDefault(innerNode => innerNode.Name == "Camera");
+                    initialCameraData = HelixWatch3DViewModel.DeserializeCamera(cameraElement);
+                }
             }
             catch (Exception ex)
             {
                 Log(LogMessage.Error(ex));
                 Log("View attributes could not be read from the file.");
+                initialCameraData = new CameraData();
```
Comment (ramramps) [inline src/Libraries/DynamoWatch3D/Watch3D.cs]: Is it ok to move initialCameraData = new CameraData() to the start, instead of doing it in Exception?

**Comment by holyjewsus** [comment]
@ikeough theres a bunch of places in the PR that `Convert.Double` and `float.parse` are not using InvariantCulture - do you think these could cause problems like we just discussed?

**Comment by ikeough** [comment]
@holyjewsus Good catch. I've added culture invariant parsing on the camera deserialization, and the node resizing.

**Pull request merged**